core/node/api_server/src/web3/namespaces/zks.rs

```diff
@@ -237,6 +238,14 @@ impl ZksNamespace {
         msg: H256,
         l2_log_position: Option<usize>,
     ) -> Result<Option<L2ToL1LogProof>, Web3Error> {
+        if let Some(handler) = &self.state.l2_l1_log_proof_handler {
```

Comment:
Dumb question just for my understanding: How is `l2_l1_log_proof_handler` going to be initialized? (AFAIU, `with_l2_l1_log_proof_handler()` method that sets it isn't currently used anywhere.) Will it proxy to the main node on ENs?

Reply:
yes, it will proxy to the main node on ENs (that's the only usecase)